api/spec/controllers/spree/api/users_controller_spec.rb

```diff
@@ -9,145 +9,164 @@ module Spree
+          expect(response.status).to eq(201)
+          expect(json_response['bill_address']['city']).to eq 'New York'
+          expect(json_response['ship_address']['city']).to eq 'Chicago'
+        end
+
+        # there's no validations on LegacyUser?
+        xit "cannot create a new user with invalid attributes" do
+          api_post :create, :user => {}, token: user.spree_api_key
+          expect(response.status).to eq(422)
+          expect(json_response["error"]).to eq("Invalid resource. Please fix errors and try again.")
+          errors = json_response["errors"]
+        end
+      end
+
+      describe "#update" do
+        it "can update own details" do
+          api_put :update, id: user.id, token: user.spree_api_key, user: {
+            email: "mine@example.com",
+            bill_address_attributes: build(:address, city: "New York").attributes,
+            ship_address_attributes: build(:address, city: "Chicago").attributes,
+          }
+          expect(json_response['email']).to eq 'mine@example.com'
+          expect(json_response['bill_address']['city']).to eq 'New York'
+          expect(json_response['ship_address']['city']).to eq 'Chicago'
+        end
+
+        it "cannot update other users details" do
+          api_put :update, id: stranger.id, token: user.spree_api_key, user: { :email => "mine@example.com" }
+          assert_not_found!
+        end
+      end
+
+      describe "#destroy" do
+        it "can delete itself" do
+          api_delete :destroy, id: user.id, token: user.spree_api_key
+          expect(response.status).to eq(204)
+        end
+
+        it "cannot delete other user" do
+          api_delete :destroy, id: stranger.id, token: user.spree_api_key
+          assert_not_found!
+        end
+
+        context "with the ability to manage users" do
+          before { stub_authentication! }
+          sign_in_as_admin!
+
+          it "can destroy user without orders" do
+            user.orders.destroy_all
+            api_delete :destroy, :id => user.id
+            expect(response.status).to eq(204)
+          end
+
+          it "cannot destroy user with orders" do
+            create(:completed_order_with_totals, :user => user)
+            api_delete :destroy, :id => user.id
+            expect(json_response["exception"]).to eq "Spree::Core::DestroyWithOrdersError"
+            expect(response.status).to eq(422)
+          end
+        end
+      end
+
+      describe "#index" do
+        it "should only get own details on index" do
+          2.times { create(:user) }
+          api_get :index, token: user.spree_api_key
+
+          expect(Spree.user_class.count).to eq 3
+          expect(json_response['count']).to eq 1
+          expect(json_response['users'].size).to eq 1
+        end
+
+        context "with the ability to display users" do
+          before do
+            user.spree_roles.create!(name: "user_display")
+            Spree::RoleConfiguration.configure do |config|
```

Comment:
hm yea. you have more knowledge here. would it be beneficial to clear these after every test run like we do for preferences? or is state required for some specs.